src/components/pages/Onboarding/Onboarding.tsx

```diff
@@ -73,13 +72,25 @@ export class OnboardingView extends React.Component<
     this.steps = [];
     this.state = {
       step: OnboardingStep.SPLASH,
-      invitingMember: this.props.deeplinkInvitingMember,
-      videoDownloadUrl: this.props.deeplinkVideoDownloadUrl
+      invitingMember: this.props.deeplinkInvitingMember
     };
   }
+
+  initializeDeeplinkingParams = async () => {
+    const videoDownloadUrl = await extractDeeplinkVideoUrl(
+      this.props.deeplinkInvitingMember,
+      this.props.deeplinkVideoToken
+    );
+    if (!videoDownloadUrl) {
+      return;
+    }
+    this.setState({ videoDownloadUrl });
+  };
+
   componentDidMount() {
     this._checkValidDeeplink();
     BackHandler.addEventListener("hardwareBackPress", this._handleBackPress);
+    this.initializeDeeplinkingParams();
```

Comment:
_checkValidDeeplink shows a dropdown if the user has an invalid deeplink -- maybe we should do that after initializeDeeplinkingParams returns the download URL? If there is no download URL we should show a dropdown.